### PR TITLE
Add author date when showing author name to librarians

### LINKED ISSUE #7799: Display author date(s) with name for librarians
Author disambiguation is difficult and even more difficult when the author names all look the same for different identities. To see precisely which author is associated with a work, one must either click the author to view the profile or go into edit mode and use the author selection box.

### Describe the problem that you'd like solved

Authors should display their associated years to make this task less cumbersome. 

### Proposal & Constraints

Only the year is necessary, any other date information does not need to be displayed.
This should be a librarian-feature only.

### Additional context

For example, these are two different Stephen Kings:
Screenshot 2023-04-19 at 09 02 16
Screenshot 2023-04-19 at 09 02 38

A display like this would make disambiguation much easier and less time-consuming:
Screenshot 2023-04-19 at 09 04 35
Screenshot 2023-04-19 at 09 04 47

### PR DESCRIPTION
**Pull request by Aeltumn**
Closes #7799

[feature]

### Technical

This is a continuation of #8076. Addresses the comments on this previous PR and completes the feature. Notable changes:
- Removed integration test as those have since been removed
- Don't show the ( - ) if no birth date is known. It should show when no death date is set, but not if there's no birth date either
- Use a dictionary to store the author information and only pass along the dates on the book view
- Only show the years if the user is in librarian mode
- Added a regex to determine the year, using one to just take the first 4 consequtive digits. This is based on a comment by @cdrini. Is this a fitting regex or did you have another one in mind?

It took some effort to figure out use to run the regex as it needs the re import. So there might have been a simpler way.

### Testing

Log in with an account that is in `/usergroup/librarian`.
Navigate to a random book and check if the author is showing a birth and death year.

### Screenshot

Before logging in
loggedout

After logging in
loggedin

### Stakeholders

@jimchamp

### CODE REVIEW

File: openlibrary/templates/books/edit.html [hunk 1]
```diff
@@ -58,7 +58,7 @@ <h2 class="editFormTitle">$:title</h2>
         <h1 class="editFormBookTitle">$this_title</h1>
         $if authors:
             <h3 class="editFormBookAuthors">
-                $:macros.BookByline([(author.name, author.url(), author.birth_date, author.death_date) for author in authors])
+                $:macros.BookByline([{'name': a.name, 'url': a.url()} for author in authors])
```
Comment (jimchamp) [inline openlibrary/templates/books/edit.html:61]:
```suggestion
                $:macros.BookByline([{'name': a.name, 'url': a.url()} for a in authors])
```
This is causing errors in book edit pages.  Did you test this?  Also, the birth and death dates would be useful to librarians while they are editing.

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks, @Aeltumn!

This should be good to go after the new suggested changes are implemented.

File: openlibrary/templates/books/edit.html (hunk 1 above)
Comment (Aeltumn) [reply openlibrary/templates/books/edit.html:61]: Managing my changes in git was a bit of a mess, so I must've gotten this mixed up somewhere.
Will fix, add the birth/date date here, and test.

File: openlibrary/templates/type/edition/title_and_author.html [hunk 2]
```diff
@@ -37,11 +37,12 @@ <h2 class="work-subtitle">
     $ author_names = (work or edition).author_names
     $if authors:
       <h2 class="edition-byline">
-        $:macros.BookByline([(author.name, author.url(), author.birth_date, author.death_date) for author in authors], attrs='itemprop="author"')
+        $ is_librarian = ctx.user and (ctx.user.is_librarian() or ctx.user.is_super_librarian())
```
[the remark this answers is not in the document]
Comment (Aeltumn) [reply openlibrary/templates/type/edition/title_and_author.html:40]: I copied the entire statement from another template, hence the or super_librarian. But I'll delete it from here after I'm done testing.

**Comment by Aeltumn** [comment]
I've addressed the comments and re-tested my changes.

**Review by jimchamp — APPROVED** [review]
Everything seems to be working as expected on testing.  Thanks, @Aeltumn.

**Pull request merged**